docs/book/src/tasks/experimental-features/runtime-sdk/implement-topology-mutation-hook.md

```diff
@@ -192,6 +296,97 @@ function openSwaggerUI() {
 }
 </script>
 
+### DiscoverVariables
+
+A DiscoverVariables call returns definitions for one or more variables. This variable definition is the same as that used for inline variables.
+
+Example variable definition:
+
+```yaml
+  - name: etcdImageTag # name of the variable
+    required: true # whether the variable must be defined for Clusters.
+    schema:
+      openAPIV3Schema: 
+        type: string 
+        default: "3.5.3-0" # variable value which is defaulted during reconciliation if not defined elsewhere.
+        example: "3.5.3-0" # example for user guidance.
+        description: "etcdImageTag sets the tag for the etcd image." #description of variable use.
+        # Additional fields from clusterv1.JSONSchemaProps
+```
+
+#### Example Request:
+
+* The request is a simple request which contains the name and namespace of the ClusterClass it is called for.
+TODO: Should this send the full ClusterClass?
```

Comment:
I think not, the name should be ok.
Also, I think we are missing the settings

Reply:
Echoing what Fabrizio said, we should add settings here. Settings is part of all runtime extension hook request objects.

Reply:
As mentioned here: https://github.com/killianmuldoon/cluster-api/pull/48/files#r1065849823

I"m not sure if we should send the ClusterClass name. I think we facilitate some strange use cases this way.

But no strong objection from my side if we think it's a good idea.